examples/go/tagging/handler/handler.go

```diff
@@ -17,58 +17,56 @@ import (
 	"github.com/vmware/govmomi/vim25/types"
 )
 
-// tagInfo is part of the toml vcconfig file
-type tagInfo struct {
-	Action string
-	Urn    string
-}
-
-// VCenter is part of the toml vcconfig file
-type vCenterAuth struct {
-	Password string
-	Server   string
-	User     string
-}
-
 // Config represents the toml vcconfig file
 type vcConfig struct {
-	Tag     tagInfo
-	VCenter vCenterAuth
+	Tag struct {
+		Action string
+		Urn    string
+	}
+	VCenter struct {
+		Password string
+		Server   string
+		User     string
+	}
 }
 
-// The request from vmware-event-router.
-type dataObj struct {
-	Data nest1 `json:"data"`
+// The structure for a Cloud Event.
```

Comment:
These structures still throws me off when reviewing the code, especially since you use a different json tag for the "data" field (below). Since you already import the govmomi types, I think it makes sense to also leverage the `events.go` package (from the VMware Event Router folder) or copy the structure over to this file.

Reply:
If you want a clean and simple approach (i.e. you only care about moref and nothing else in the CloudEvents payload):

```go
type incoming struct {
	Data types.Event `json:"data,omitempty"`
}
```

Then you need to check whether it's a proper event carrying vm information (note that not only VmEvent* can carry vm information, so the user must make sure to put in the correct event topics for this function, but this applies to all fn in general).

Pseudo code:

```go
var event incoming
	err = json.Unmarshal(b, &event)
	if err != nil {
		log.Println(err) // handle invalid JSON
                return
	}

if event.Data == nil || event.Data.Vm == nil {
	log.Println("is nil") // handle invalid VmEvent payload
	return
}
log.Printf("moref: %v", event.Data.Vm.Vm)
```